File: python/unions/hello/hello_stack.py
from aws_cdk import core
import aws_cdk.aws_sam as sam


class HelloStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        fun = sam.CfnFunction(self, 'Fn',
            code_uri='asdf',
            handler='index.handler',
            runtime='runtime',
            events={
              'Event' : sam.CfnFunction.EventSourceProperty(
                type='type1',
                properties=sam.CfnFunction.SQSEventProperty(queue='queue-arn'))
              })

        print(fun.events['Event'].properties)
        assert fun.events['Event'].properties.queue == 'queue-arn'

File: java/pass-struct-through-union/src/main/java/com/myorg/HelloStack.java
package com.myorg;

import software.amazon.awscdk.core.Construct;
import software.amazon.awscdk.core.Stack;
import software.amazon.awscdk.core.StackProps;
import software.amazon.awscdk.services.sam.CfnFunction;

import java.util.HashMap;
import java.util.Map;

public class HelloStack extends Stack {
    public HelloStack(final Construct parent, final String id) {
        this(parent, id, null);
    }

    public HelloStack(final Construct parent, final String id, final StackProps props) {
        super(parent, id, props);

        CfnFunction fun = CfnFunction.Builder.create(this, "Fn")
                .codeUri("asdf")
                .handler("index.handler")
                .runtime("runtime")
                .events(new HashMap<String, Object>() {{
                    put("Event", CfnFunction.EventSourceProperty.builder()
                            .type("type1")
                            .properties(CfnFunction.SQSEventProperty.builder()
                                    .queue("queue-arn")
                                    .build()
                            )
                            .build()
                    );
                }})
                .build();

//        Map<String, CfnFunction.EventSourceProperty> events = (Map<String, CfnFunction.EventSourceProperty>) fun.getEvents();
//        System.out.println(events.get("Event").getProperties());
    }
}
